### PR TITLE
fix(cli/worker): Print error stacks from the origin Worker

### LINKED ISSUE #4728: Error stack in worker missing
Currently, every error thrown in worker has no trusted stacks, but only
```
at WorkerImpl.#poll ($deno$/web/workers.ts:140:17)
```

### PR DESCRIPTION
**Pull request by nayeemrmn**
Fixes #4728.

cc @bartlomieju

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
I think this looks great and we could land it, but want to get another pair of eyes on it

File: cli/ops/worker_host.rs [hunk 1]
```diff
@@ -40,6 +51,21 @@ pub fn init(rt: &mut deno_core::JsRuntime) {
   );
   super::reg_json_sync(rt, "op_host_post_message", op_host_post_message);
   super::reg_json_async(rt, "op_host_get_message", op_host_get_message);
+  super::reg_json_sync(
+    rt,
+    "op_host_unhandled_error",
+    move |_state, args, _zero_copy| {
```
Comment (kitsonk) [inline cli/ops/worker_host.rs:57]: Was there a reason to not be consistent with the other ops and create a seperate function?

File: cli/ops/worker_host.rs [hunk 2]
```diff
@@ -40,6 +51,21 @@ pub fn init(rt: &mut deno_core::JsRuntime) {
...
+      if let Some(mut sender) = sender.clone() {
+        let args: HostUnhandledErrorArgs = serde_json::from_value(args)?;
+        sender
+          .try_send(WorkerEvent::Error(generic_error(args.message)))
+          .expect("Failed to propagate error event to parent worker");
+        Ok(json!({}))
```
Comment (kitsonk) [inline cli/ops/worker_host.rs:63]: In other cases, we respond with `json!(true)` for a "non" response.

File: cli/tests/subdir/worker_error.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+function foo() {
+  throw new Error("foo");
```
Comment (kitsonk) [inline cli/tests/subdir/worker_error.ts:2]: What happens with other built-in error classes like `TypeError`, `RangeError`, `SyntaxError`?  Do we need to send more than just the error message?

**Review by kitsonk — COMMENTED** [review]
A couple thoughts.

File: cli/ops/worker_host.rs (hunk 1 above)
Comment (nayeemrmn) [reply cli/ops/worker_host.rs:57]: It's a closure that uses structures only safe (or convenient) to retrieve during init, not available in op context under current infrastructure. Suggested here: https://discord.com/channels/684898665143206084/684911491035430919/766354819925278750.

File: cli/tests/subdir/worker_error.ts (hunk 3 above)
Comment (nayeemrmn) [reply cli/tests/subdir/worker_error.ts:2]: The "message" being sent is `Uncaught <error.name>: <error.message>`. This is all that's sent to the host in browsers: 
image

File: cli/ops/worker_host.rs (hunk 2 above)
Comment (nayeemrmn) [inline cli/ops/worker_host.rs:63]:
```suggestion
        Ok(json!(true))
```

File: cli/tests/subdir/worker_error.ts (hunk 3 above)
Comment (kitsonk) [reply cli/tests/subdir/worker_error.ts:2]: Ok, cool!

**Review by kitsonk — APPROVED** [review]
LGTM!

**Pull request merged**